utils/runtime-fuzzer/src/gear_calls.rs

```diff
@@ -0,0 +1,359 @@
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! `GearCalls` implementation.
+
+use crate::{GearCall, SendMessageArgs, UploadProgramArgs};
+use arbitrary::{Error, Result, Unstructured};
+use gear_call_gen::{ClaimValueArgs, SendReplyArgs};
+use gear_core::ids::{CodeId, MessageId, ProgramId};
+use gear_utils::NonEmpty;
+use gear_wasm_gen::{
+    EntryPointsSet, ParamType, StandardGearWasmConfigsBundle, SysCallName,
+    SysCallsInjectionAmounts, SysCallsParamsConfig,
+};
+
+/// Maximum payload size for the fuzzer - 512 KiB.
+const MAX_PAYLOAD_SIZE: usize = 512 * 1024;
+static_assertions::const_assert!(MAX_PAYLOAD_SIZE <= gear_core::message::MAX_PAYLOAD_SIZE);
+
+/// The struct for generating gear extrinsics.
+///
+/// # Usage
+///
+/// `Iterator<Item = GearCall>` is implemented for this struct, so for
+/// generating gear calls you need to iterate over [`GearCalls`].
+pub(crate) struct GearCalls<'a> {
+    unstructured: Unstructured<'a>,
+    intermediate_data: IntermediateData,
+
+    config: Config,
+
+    current_generator: usize,
+    current_extrinsic: usize,
+}
+
+impl<'a> GearCalls<'a> {
+    pub fn new(data: &'a [u8], config: Config) -> Result<GearCalls<'a>> {
+        if data.len() < config.unstructured_size_hint() {
+            return Err(Error::NotEnoughData);
+        }
+
+        Ok(GearCalls {
+            unstructured: Unstructured::new(data),
+            intermediate_data: IntermediateData::default(),
+            current_generator: 0,
+            current_extrinsic: 0,
+            config,
+        })
+    }
+}
+
+impl Iterator for GearCalls<'_> {
+    type Item = Result<GearCall>;
+
+    fn next(&mut self) -> Option<Result<GearCall>> {
+        if self.current_generator == self.config.generators.len() {
+            return None;
+        }
+
+        let (_, generator) = &self.config.generators[self.current_generator];
+        let call = generator.generate(&mut self.intermediate_data, &mut self.unstructured);
+
+        self.current_extrinsic += 1;
+        if self.current_extrinsic == self.config.generators[self.current_generator].0 {
+            self.current_extrinsic = 0;
+            self.current_generator += 1;
+        }
+
+        Some(call)
+    }
+}
+
+pub(crate) type ExtrinsicAmount = usize;
+
+/// Config that's used in the process of generation gear calls.
+pub(crate) struct Config {
```

Comment:
but it have sense in the context of gear_calls::Config

Reply:
Your type doesn't hold any configurational data which is used to instantiate main entities of the program or change it workflow.

To be honest, it really looks like "config", I'd say indirectly, because you really use some data of the type to instantiate generators. But, anyway, that's not enough for calling it config.